### DISCUSSION TITLE
Generic using alias directives

### DISCUSSION THREAD

**Discussion by naine** [dotnet/csharplang#4259] [Language Ideas] [answered]
Currently it is possible to create file-scoped type aliases in C#.

```csharp
using Foo = System.Collections.ArrayList;
```

For generic types, it is currently only possible to create aliases to constructed types.

```csharp
using Foo = System.Collections.Generic.List<int>;
```

I propose to allow the alias itself to be generic, using syntax like the following:

```csharp
using Foo<T> = System.Collections.Generic.List<T>;
using Bar<T> = System.Collections.Generic.Dictionary<string, T>;
using Bar<T1, T2> = System.Collections.Generic.Dictionary<T1, T2>;
```

**Comment by HaloFour** [comment, marked as answer]
See: https://github.com/dotnet/csharplang/issues/1239
